### PR TITLE
[WFLY-15016] Reactive Messaging and AMQ Streams support

### PR DESCRIPTION
**Pull request by kabir**
https://issues.redhat.com/browse/WFLY-15016

### CODE REVIEW

File: microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+* https://issues.redhat.com/browse/WFLY-14932[WFLY-14932 - Configuration of Reactive Messaging Kafka messages] - The API introduced here allows you to get information about how Kafka handled a message, and to influence how messages are sent to Kafka. Most importantly it enables the use of a key for the messages sent to Kafka, which in turn inside Kafka are stored as records. Being able to query them, and not simply use Kafka as a messaging system is important.
+* https://issues.redhat.com/browse/WFLY-14987[WFLY-14987 - Ability to use an Elytron SSL Context in the Kafka connector] - The Kafka instance set up by AMQ Streams might be secured using SSL with unsigned certificates, so we need a way to specify the truststore.
+* https://issues.redhat.com/browse/WFLY-13640[WFLY-13640 - Provide Support for MicroProfile Reactive Messaging 1.0 for Kafka] - This was the original issue bringing Reactive Messaging 1.0 into WildFly. Some testing was done against AMQ Streams on OpenShift.
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+TBD
```
Comment (fabiobrz) [inline microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc:32]: We can exceptionally add 3 contacts, BTW :smile: - Jan Stourac, Miroslav Novak and myself.

File: microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+** `PLAINTEXT` - Communication is not encrypted and no autentication is needed
+* `SASL_PLAINTEXT` - No encryption but with SASL-based authentication
+* `TLS` - Communication is encrypted via either a CA or unsigned certificate. AMQ Streams supports the use of client certificates for optional authentication with this mechanism
+* `SASL_SSL` - Encrypted communication with SASL-based authentication
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+TBD as we identify pain points during testing. I am not intimately familiar with Kafka/AMQ Streams.
```
Comment (fabiobrz) [inline microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc:71]: Any updates to resolve this section? Anything we should mention? BTW I see there's an open bug affecting the implementation, i.e. https://issues.redhat.com/browse/WFWIP-367

File: microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+== Test Plan
+For WildFly we test what we can as indicated in the linked <<Related Issues>>.
+
+== Community Documentation
+This is covered by the linked <<Related Issues>>.
+
+== Release Note Content
+////
```
Comment (fabiobrz) [inline microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc:80]: This proposal is mostly about testing and documentation, so just asking: are there no community release notes that can be mentioned here? Maybe those of the related issues here too?

**Review by fabiobrz — COMMENTED** [review]
Hi @kabir - I had a quick look and dropped a couple of comments and questions.
Feel free to let me know what you think.

File: microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15016[WFLY-15016]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-14932[WFLY-14932 - Upgrade to MicroProfile Reactive Messaging 2.0] - This version introduces the new `@Channel` annotation, which makes it easier for users to get data into/from the Reactive Messaging Streams
```
Comment (fabiobrz) [inline microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc:22]: Wrong issue here, it should be the following one: https://issues.redhat.com/browse/WFLY-14798

File: microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+WildFly
```
Comment (fabiobrz) [inline microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc:42]: Given what we discussed recently about the involved components, shouldn't we add SmallRye RM too? Just asking, not sure it fits here.

**Review by fabiobrz — CHANGES_REQUESTED** [review]
Hi @kabir - I dropped a couple more comments, so maybe we can have this proposal approved.

File: microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc (hunk 2 above)
Comment (kabir) [reply microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc:71]: This one is out of our hands really, as mentioned in https://issues.redhat.com/browse/WFWIP-367?focusedId=15583303&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15583303

File: microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc (hunk 3 above)
Comment (kabir) [reply microprofile/WFLY-15016-mp-reactive-messaging-amq-streams-support.adoc:80]: Not really, connecting to AMQ Streams is essentially the same as we do to connect to Kafka. So there isn't anything special to work with AMQ Streams. Also the quickstart https://github.com/wildfly/quickstart/tree/main/microprofile-reactive-messaging-kafka covers how to set up AMQ Streams and connecting to it  from WildFly

**Pull request closed without merging**